src/borg/archiver/debug_cmd.py

```diff
@@ -265,6 +267,74 @@ def do_debug_id_hash(self, args, repository, manifest):
         print(id.hex())
         return EXIT_SUCCESS
 
+    @with_repository(compatibility=Manifest.NO_OPERATION_CHECK)
+    def do_debug_parse_obj(self, args, repository, manifest):
+        """parse object file into meta dict and data (decrypting, decompressing)"""
+
+        # get the object from id
+        hex_id = args.id
+        try:
+            id = unhexlify(hex_id)
+            if len(id) != 32:  # 256bit
+                raise ValueError("id must be 256bits or 64 hex digits")
+        except ValueError as err:
+            print("object id %s is invalid [%s]." % (hex_id, str(err)))
+            return EXIT_ERROR
+
+        with open(args.cdata, "rb") as f:
+            cdata = f.read()
+
+        repo_objs = manifest.repo_objs
+
+        try:
+            meta, data = repo_objs.parse(id=id, cdata=cdata, decompress=True)
+        except Exception:
+            traceback.print_exc()
+            return EXIT_ERROR
```

Comment:
iirc, if you do not catch the exception, it is borg's default behaviour to print a traceback and exit with EXIT_ERROR.